### DISCUSSION TITLE
RichLog write without new line at the end

### DISCUSSION THREAD

**Discussion by mattthhh** [Textualize/textual#4289] [Q&A] [answered]
Hello ! I wanted to know if it's possible to write in the RichLog but without printing a new line at the end ? Thanks in advance :)

**Comment by davep** [comment, marked as answer]
To the best of my recollection `RichLog` doesn't support that, although `Log` does.

**Comment by huiyeruzhou** [comment]
@on(events.Print)
    def log_printed(self, event: events.Print):
        # time:``
        if not event.text: #####################ignore empty line
            return
        time_str = datetime.datetime.now().strftime('%H:%M:%S')
        self.query_one(".logger", RichLog).write(f"[{time_str}] {event.text}")
  This work for me when I'm using self.begin_capture_print(self, True, True) to catch log.
